packages/angular/src/schematics/ngrx/rules/add-imports-to-module.ts

```diff
@@ -25,105 +33,136 @@ export function addImportsToModule(context: RequestContext): Rule {
 
     // InsertImport [symbol,source] value pairs
     const nxModuleImport = ['NxModule', '@nrwl/angular'];
-    const storeModule = ['StoreModule', '@ngrx/store'];
-    const effectsModule = ['EffectsModule', '@ngrx/effects'];
-    const storeDevTools = ['StoreDevtoolsModule', '@ngrx/store-devtools'];
-    const environment = ['environment', '../environments/environment'];
-    const storeRouter = ['StoreRouterConnectingModule', '@ngrx/router-store'];
-    const storeFreeze = ['storeFreeze', 'ngrx-store-freeze'];
 
     // this is just a heuristic
     const hasRouter = sourceText.indexOf('RouterModule') > -1;
     const hasNxModule = sourceText.includes('NxModule.forRoot()');
 
-    if (context.options.onlyEmptyRoot) {
+    const setupNx = () => {
       insert(host, modulePath, [
-        addImport.apply(this, storeModule),
-        addImport.apply(this, effectsModule),
-        addImport.apply(this, storeDevTools),
-        addImport.apply(this, environment),
-        ...(hasRouter ? [addImport.apply(this, storeRouter)] : []),
-        addImport.apply(this, storeFreeze),
-        ...addImportToModule(source, modulePath, storeForEmptyRoot),
-        ...addImportToModule(source, modulePath, effectsForEmptyRoot),
-        ...addImportToModule(source, modulePath, devTools),
-        ...(hasRouter
-          ? addImportToModule(source, modulePath, storeRouterModule)
-          : [])
+        ...(!hasNxModule ? [addImport.apply(this, nxModuleImport)] : []),
+        ...(!hasNxModule ? addImportToModule(source, modulePath, nxModule) : [])
       ]);
-    } else {
-      let common = [
-        addImport.apply(this, storeModule),
-        addImport.apply(this, effectsModule),
-        addImport(reducerImports, reducerPath),
-        addImport(effectsName, effectsPath)
-      ];
-      if (context.options.facade) {
-        common = [
-          ...common,
-          addImport(facadeName, facadePath),
-          ...addProviderToModule(source, modulePath, `${facadeName}`)
-        ];
+
+      return noop();
+    };
+
+    const setupFacade = () => {
+      insert(host, modulePath, [
+        addImport(facadeName, facadePath),
+        ...addProviderToModule(source, modulePath, `${facadeName}`)
+      ]);
+
+      return noop();
+    };
+
+    const moduleName = basename(context.options.module as Path);
+    const relativeModuleName = `${path.relative(
+      `${context.moduleDir}/${context.options.directory}`,
+      context.moduleDir
+    )}/${moduleName}`;
+    const moduleDir = `${context.moduleDir}/${context.options.directory}`;
+
+    /**
+     * Delegate the store setup to NgRx schematics
+     */
+    const setupRootStore = externalSchematic('@ngrx/schematics', 'store', {
+      minimal: true,
+      module: relativeModuleName,
+      path: moduleDir,
+      root: true
+    });
+
+    /**
+     * Delegate the effects setup to NgRx schematics
+     */
+
+    const setupRootEffects = externalSchematic('@ngrx/schematics', 'effect', {
```

Comment:
Setting up root Effects seems to cause `app.module.ts` to become malformed:

![image](https://user-images.githubusercontent.com/8104246/61177091-ce86d400-a59a-11e9-9371-66c54b02260e.png)

Reply:
The AppFacade thing is weird. I'll look into it
The `.forRoot()` and `.forFeature()` is intentional. This is to satisfy the existing use case of registering the root with a feature. In general, we want to encourage developers to keep the root object empty, and use `forFeature()` to register additional feature states.